pkg/pool-weighted/contracts/test/ManagedPoolStorageLib.t.sol

```diff
@@ -0,0 +1,198 @@
+        assertEq(clearedOldState, clearedNewState);
+    }
+
+    function testSetRecoveryMode(bytes32 poolState, bool enabled) public {
+        bytes32 newPoolState = ManagedPoolStorageLib.setRecoveryModeEnabled(poolState, enabled);
+        assertOtherStateUnchanged(poolState, newPoolState, _RECOVERY_MODE_OFFSET, 1);
+
+        assertTrue(ManagedPoolStorageLib.getRecoveryModeEnabled(newPoolState) == enabled);
+    }
+
+    function testSwapsEnabled(bytes32 poolState, bool enabled) public {
+        bytes32 newPoolState = ManagedPoolStorageLib.setSwapsEnabled(poolState, enabled);
+        assertOtherStateUnchanged(poolState, newPoolState, _SWAP_ENABLED_OFFSET, 1);
+
+        assertTrue(ManagedPoolStorageLib.getSwapsEnabled(newPoolState) == enabled);
+    }
+
+    function testLPAllowlistEnabled(bytes32 poolState, bool enabled) public {
+        bytes32 newPoolState = ManagedPoolStorageLib.setLPAllowlistEnabled(poolState, enabled);
+        assertOtherStateUnchanged(poolState, newPoolState, _MUST_ALLOWLIST_LPS_OFFSET, 1);
+
+        assertTrue(ManagedPoolStorageLib.getLPAllowlistEnabled(newPoolState) == enabled);
+    }
+
+    function testWeightChangeFields(
+        bytes32 poolState,
+        uint32 startTime,
+        uint32 endTime
+    ) public {
+        bytes32 newPoolState = ManagedPoolStorageLib.setWeightChangeData(poolState, startTime, endTime);
+        assertOtherStateUnchanged(poolState, newPoolState, _WEIGHT_START_TIME_OFFSET, _TIMESTAMP_WIDTH * 2);
+
+        (uint256 recoveredStartTime, uint256 recoveredEndTime) = ManagedPoolStorageLib.getWeightChangeFields(
+            newPoolState
+        );
+        assertEq(recoveredStartTime, startTime);
+        assertEq(recoveredEndTime, endTime);
+    }
+
+    function testWeightChangeProgress(
+        bytes32 poolState,
+        uint32 startTime,
+        uint32 endTime,
+        uint32 now
+    ) public {
+        hevm.warp(now);
+        hevm.assume(startTime <= endTime);
+
+        bytes32 newPoolState = ManagedPoolStorageLib.setWeightChangeData(poolState, startTime, endTime);
+        uint256 weightChangeProgress = ManagedPoolStorageLib.getGradualWeightChangeProgress(newPoolState);
+
+        if (now >= endTime) {
+            assertEq(weightChangeProgress, FixedPoint.ONE);
+        } else if (now <= startTime) {
+            assertEq(weightChangeProgress, 0);
+        } else {
+            uint256 expectedWeightChangeProgress = FixedPoint.divDown(now - startTime, endTime - startTime);
+            assertEq(weightChangeProgress, expectedWeightChangeProgress);
+        }
+    }
+
+    function testSwapFeeData(
+        bytes32 poolState,
+        uint32 startTime,
+        uint32 endTime,
+        uint64 startSwapFeePercentage,
+        uint64 endSwapFeePercentage
+    ) public {
+        hevm.assume(startSwapFeePercentage < _MAX_SWAP_FEE);
+        hevm.assume(endSwapFeePercentage < _MAX_SWAP_FEE);
```

Comment:
Don't we also have a minimum fee?

Reply:
Also this should be <=

Reply:
Not for the purposes of `ManagedPoolStorageLib`. `_MAX_SWAP_FEE` here is over 100% as we're just defining the range of inputs which can be written into the slot.